lib/GHCup/Utils/Parsers.hs

```diff
@@ -379,9 +360,40 @@ fromVersion' SetNext tool = do
     GHCup -> fail "GHCup cannot be set"
   let vi = getVersionInfo next tool dls
   pure (next, vi)
-fromVersion' (SetToolTag t') tool =
+fromVersion' (SetToolTag t') _ tool =
   throwE $ TagNotFound t' tool
 
+guessFullVersion :: ( HasLog env
+                    , MonadFail m
+                    , MonadReader env m
+                    , HasGHCupInfo env
+                    , HasDirs env
+                    , MonadThrow m
+                    , MonadIO m
+                    , MonadCatch m
+                    )
+                 => GHCTargetVersion
+                 -> Tool
+                 -> GuessMode
+                 -> m (GHCTargetVersion, Maybe VersionInfo)
+guessFullVersion v tool guessMode = do
+  GHCupInfo { _ghcupDownloads = dls } <- getGHCupInfo
+  let vi = getVersionInfo v tool dls
+  case pvp $ prettyVer (_tvVersion v) of -- need to be strict here
+    Left _ -> pure (v, vi)
+    Right pvpIn
+      | (guessMode /= GStrict) && hasn't (ix tool % ix v) dls -> do
+          ghcs <- if guessMode == GLaxWithInstalled then fmap rights getInstalledGHCs else pure []
```

Comment:
need to do `getInstalledCabals` etc here?

Reply:
Done